AsyncDisplayKit/ASNetworkImageNode.mm

```diff
@@ -246,6 +280,9 @@ - (void)visibilityDidChange:(BOOL)isVisible
           }
           
           strongSelf.image = progressImage;
+          ASPerformBlockOnMainThread(^{
+            strongSelf->_currentImageQuality = progress;
```

Comment:
This and other trampolined blocks I believe are un-locked accesses; is that permissible?  It would have some overhead to grab the lock of course, but it may be necessary to go through the property setter.  I wonder if there is any deadlock risk there.